fire/cli/ts/hts.py

```diff
@@ -206,3 +208,198 @@ def plot_hts(objekt: str, plottype: str, parametre: str, minpunkter, **kwargs) -
+    \b
+        ki_hældning_        Nedre/øvre grænse for konfidensinterval for estimeret
+                            hældning. Konfidensintervallet er bestemt ved
+                            signifikansniveau på 5%
+    \b
+        mex                 Middelepoke for tidsserien (Gennemsnit af x-værdier)
+    \b
+        mey                 Tidsseriens fittede værdi ved middelepoken
+    \b
+        er_bevægelse_signifikant    Resultat af hypotesetest (T-test) for om
+                                    bevægelsen er signifikant forskellig fra 0
+    \b
+        alpha_bevægelse_signifikant Signifikansniveau anvendt i T-test (default 1%)
+
+    I de detaljerede plots vises som nævnt konfidensbånd for fittet. Hertil anvendes
+    signifikansniveau på 5%.
+    """
+    # skaler data så der regnes og plottes i [mm] i stedet for [m]
+    skalafaktor = 1e3
+
+    # Minimum spredning
+    apriori_spredning = 0.5 # [mm]
+
+    # Hent tidsserier som skal analyseres baseret på bruger input.
+    try:
+        # Antag først at der er givet en punktsamling
+        punktsamling = fire.cli.firedb.hent_punktsamling(objekt[0])
+    except NoResultFound:
+        # Ellers må det være tidsserier
+        tidsserier = (
+            fire.cli.firedb.session.query(HøjdeTidsserie)
+            .filter(
+                HøjdeTidsserie._registreringtil == None,
+                HøjdeTidsserie.navn.in_(objekt),
+            )
+            .all()
+        )  # NOQA
+
+        if not tidsserier:
+            raise SystemExit("Fandt ingen tidsserier")
+
+        punktsamling = tidsserier[0].punktsamling
+        punktsamling.navn
+        # Tjek at tidsserierne alle har samme jessenpunkt og jessenkote
+        ugyldige_tidsserier = [
+            ts
+            for ts in tidsserier
+            if not (ts.punktsamling.jessenkote == punktsamling.jessenkote
+            and ts.punktsamling.jessenpunkt == punktsamling.jessenpunkt)
+        ]
+
+        if ugyldige_tidsserier:
+            raise SystemExit("Fandt tidsserier i forskellige lokale højdesystemer. Afbryder!")
+
+    else:
+        tidsserier: list[HøjdeTidsserie] = punktsamling.tidsserier
+
+    # Filtrér desuden på minimum antal punkter, da de ikke kan filtreres via SQL
+    tidsserier = [
+        ts
+        for ts in tidsserier
+        if (len(ts) >= nmin )
+    ]
+
+    if not tidsserier:
+        raise SystemExit("Fandt ingen tidsserier")
+
+    for ts in tidsserier:
+        print(f"Fandt {ts.navn}")
+
+    # Beregn lineær regression for alle tidsserier samt statistik til rapportering.
+    ts_statistik = {}
+    ts_fejlende = []
+    for ts in tidsserier:
+        y = [skalafaktor * yy for yy in ts.kote]
+
+        # Divider med 1000, idet spredningerne allerede er i mm
+        # Anvend minimum værdi for spredning, da mange spredninger i databasen er nul
+        y_vægte = [1/(skalafaktor * max(sy, apriori_spredning)/1e3)**2  for sy in ts.sz]
```

Comment:
Spredning sættes til 0,5 selvom den måske reelt er 0,1 - udgør det noget særligt problem? Jeg forstår det er gjort for at tage hånd om de gamle informationer hvor der i REFGEO er rundet ned til 0.0, men man kan måske godt forestille sig at vi indsætte nye koter med en meget lav spredning i dag.

Reply:
Ja true. Kunne godt lave det så spredning kun sættes til `apriori_spredning`, hvis den er 0 i forvejen.

Reply:
Do it! Så slipper vi måske for mærkelige overraskelser ude i fremtiden

Reply:
har brugt `(sy if sy!=0 else apriori_spredning)` i stedet for `max(sy, apriori_spredning)`